### PR TITLE
Improve handling of sub rpms.

### PR DESCRIPTION
**Pull request by aiuto**
- Do not create the subrpms files when we don't have to
- Do not add `buildsubdir .`  That seems to cause rpmbuild failures with older rpmbuild. It generates a cleanup
  ```
   rm -rf . /tmp/tmpXXXXX/BUILD
   ```
   `rm` chokes on the . and fails.
- General cleanup: Remove tests against None which is needlessly brittle.
This is a follow to #824. 
@kellyma2 : Can you test this against your setup to see if the buildsubdir change breaks you?

### CODE REVIEW

**Comment by kellyma2** [comment]
Will do.

File: pkg/make_rpm.py
Comment (aiuto) [reply pkg/make_rpm.py:369]: rpmbuild fails while cleaning up, trying to execute
`rm -rf . /tmp/tmpXXXXX/BUILD`
It chokes on the `.` 
The question is if we need `buildsubdir` in some situations or rpmbuild versions, but not others.

Comment (kellyma2) [reply pkg/make_rpm.py:369]: What version of rpmbuild was this failing and/or what distro?

Comment (kellyma2) [reply pkg/make_rpm.py:369]: Doing a bit of testing in my environment were I've included my not yet pushed changes for capturing debuginfo RPMs and it fails in that scenario.  I'll take my debuginfo changes and test it in the vanilla environment and see what happens there.

**Comment by cgrindel** [comment]
FYI. I am waiting to hear back  about @kellyma2 testing before hitting approve.

**Comment by kellyma2** [comment]
> FYI. I am waiting to hear back  about @kellyma2 testing before hitting approve.

Sorry, I think I'm ok to merge this, should have been clearer in my comments.  I'm away for the  week but have done some clearer tests around `buildsubdir` and how it needs to be used.  I'll have a PR when I'm back that covers this a bit better and also includes debuginfo support

**Review by cgrindel — APPROVED** [review]

File: pkg/make_rpm.py
Comment (aiuto) [reply pkg/make_rpm.py:369]: I'm not sure if this is an implicit approve, or if I should hold off.

**Comment by aiuto** [comment]
Silly me. Closed without merging. Necromerging now.

File: pkg/make_rpm.py
Comment (kellyma2) [reply pkg/make_rpm.py:369]: Sorry, yes, it was intended to convey approval!

Comment (aiuto) [reply pkg/make_rpm.py:369]: Got it. I merged it this morning. Now lets see if that is compatible with your work on the debuginfo.

**Pull request merged**
